app/src/main/java/com/example/myanimereport/fragments/HomeFragment.java

```diff
@@ -124,6 +183,54 @@ public void switchLayout() {
         adapter.notifyItemRangeChanged(0, entries.size());
     }
 
+    /* Filters by genre. */
+    public void filterGenres () {
+        GenreFilterBinding dialogBinding = GenreFilterBinding.inflate(getLayoutInflater());
+        List<String> genres = new ArrayList<>(allGenres);
+        genres.sort(String::compareTo);
+        genres.add(0, "All");
+
+        // Add a checkbox for each genre
+        List<CheckBox> cbs = new ArrayList<>();
+        for (String genre: genres) {
+            CheckBox cb = new CheckBox(getContext());
+            cb.setText(genre);
+            cb.setTextColor(ContextCompat.getColor(requireContext(), R.color.white));
+            if ((selectedGenres.isEmpty() && !entries.isEmpty()) || selectedGenres.contains(genre)) cb.setChecked(true);
+            dialogBinding.llGenres.addView(cb);
```

Comment:
Did you mean `dialogBinding.allGenres`?

Reply:
`llGenres` is for "linear layout genres", I'll write it out so it's less confusing